File: commons/javafx/src/main/java/com/cedarsoft/commons/javafx/HasI18nKey.java
package com.cedarsoft.commons.javafx;

import javax.annotation.Nonnull;

/**
 * Marker interface for classes the provide an i18n key.
 * That key can be used to be translated with a RES instance
 */
@FunctionalInterface
public interface HasI18nKey {
  /**
   * Returns the i18n key for this object
   */
  @Nonnull
  String getI18nKey();
}

File: commons/javafx/src/main/java/com/cedarsoft/commons/javafx/HasI18nKey.kt
package com.cedarsoft.commons.javafx

/**
 * Marker interface for classes the provide an i18n key.
 * That key can be used to be translated with a RES instance
 */
@FunctionalInterface
interface HasI18nKey {
  /**
   * Returns the i18n key for this object
   */
  val i18nKey: String
}
